packages/composables/src/composables/useWishlist/index.ts

```diff
@@ -4,38 +4,55 @@ import {
   useWishlistFactory,
   UseWishlistFactoryParams
 } from '@vue-storefront/core';
-import type { Wishlist, WishlistItem, Product } from '@vue-storefront/vendure-api';
+import type { Wishlist, WishlistItem } from '@vue-storefront/vendure-api';
+import type { AgnosticProductVariant } from '../../types';
 
-const params: UseWishlistFactoryParams<Wishlist, WishlistItem, Product> = {
+const params: UseWishlistFactoryParams<Wishlist, WishlistItem, AgnosticProductVariant> = {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   load: async (context: Context) => {
-    console.log('Mocked: useWishlist.load');
-    return {};
+    if (process.client) {
+      return JSON.parse(localStorage.getItem('wishlist')) || [];
+    }
   },
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  addItem: async (context: Context, { currentWishlist, product }) => {
-    console.log('Mocked: useWishlist.addItem');
-    return {};
+  addItem: async (context: Context, { product }) => {
+    if (!localStorage.getItem('wishlist') || localStorage.getItem('wishlist') === '1') {
+      localStorage.setItem('wishlist', JSON.stringify([]));
+    }
+    const wishlist = JSON.parse(localStorage.getItem('wishlist'));
+    wishlist?.push(product);
+    localStorage.setItem('wishlist', JSON.stringify(wishlist));
+    return wishlist;
   },
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  removeItem: async (context: Context, { currentWishlist, product }) => {
-    console.log('Mocked: useWishlist.removeItem');
-    return {};
+  removeItem: async (context: Context, { product }) => {
+    const wishlist = JSON.parse(localStorage.getItem('wishlist'));
+    const updatedWishlist = wishlist.filter(wishlistItem => wishlistItem.sku !== product.sku);
+    localStorage.setItem('wishlist', JSON.stringify(updatedWishlist));
+    return updatedWishlist;
   },
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   clear: async (context: Context, { currentWishlist }) => {
-    console.log('Mocked: useWishlist.clear');
-    return {};
+    if (process.client) {
+      localStorage.setItem('wishlist', '[]');
+    }
+    return [];
   },
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  isInWishlist: (context: Context, { currentWishlist, product }) => {
-    console.log('Mocked: useWishlist.isInWishlist');
-    return false;
+  isInWishlist: (context: Context, { product }) => {
+    const wishlist = process.client ? JSON.parse(localStorage.getItem('wishlist')) : [];
```

Comment:
Here you can create a helper that you could use globally as you are getting the wishlist object several times in this file

Reply:
Added the helpers to set and get the wishlist from localStorage. Much cleaner.